src/compiler/tsc.ts

```diff
@@ -415,7 +415,8 @@ namespace ts {
 
             const compileResult = compile(rootFileNames, compilerOptions, compilerHost);
 
-            if (!compilerOptions.watch) {
+            // Firefox has Object.prototype.watch
+            if (!compilerOptions.watch || !compilerOptions.hasOwnProperty("watch")) {

```

Comment:
The variable can change without breaking any external consumer. Just as compiler options received from the caller, legacy `watch` can be detected and translated to `watchOption`.